packages/react-core/src/components/Page/PageSidebar.tsx

```diff
@@ -19,33 +19,49 @@ export interface PageSidebarProps extends React.HTMLProps<HTMLDivElement> {
   theme?: 'dark' | 'light';
 }
 
+export interface PageSidebarContextProps {
+  className: string;
+  nav: React.ReactNode;
+  isNavOpen: boolean;
+  theme: string;
+}
+
+export const PageSidebarContext = React.createContext<Partial<PageSidebarContextProps>>({
+  className: '',
+  nav: null,
+  isNavOpen: true,
+  theme: 'dark'
+});
+
 export const PageSidebar: React.FunctionComponent<PageSidebarProps> = ({
   className = '',
   nav,
   isNavOpen = true,
   theme = 'dark',
   ...props
 }: PageSidebarProps) => (
-  <PageContextConsumer>
-    {({ isManagedSidebar, isNavOpen: managedIsNavOpen }: PageSidebarProps) => {
-      const navOpen = isManagedSidebar ? managedIsNavOpen : isNavOpen;
+  <PageSidebarContext.Provider value={{ isNavOpen }}>
```

Comment:
I would bring this down a level and wrap the `{nav}` more directly, this way you would also have the definitive answer to whether the nav is open, i.e. (leaving some code out for brevity)
```
<PageContextConsumer>
  <div id="page-sidebar">
     <div className={css(styles.pageSidebarBody)}>
        <PageSidebarContext.Provider value={{ isNavOpen: navOpen }}>
          {nav}
        </PageSidebarContext.Provider>
      </div>
  </div>
</PageContextConsumer>
```

Reply:
Thank you!! 😄